switchapi/es2k/switch_pd_routing.c

```diff
@@ -548,10 +641,18 @@ switch_status_t switch_pd_rif_mod_start_entry(
         goto dealloc_resources;
     }
 
-    status = tdi_key_field_set_value(key_hdl, field_id,
-                                     (rif_handle &
-                                     ~(SWITCH_HANDLE_TYPE_RIF <<
-                                     SWITCH_HANDLE_TYPE_SHIFT)));
+    if (SWITCH_RIF_HANDLE(rif_handle)) {
+      status = tdi_key_field_set_value(key_hdl, field_id,
+                                       (rif_handle &
+                                       ~(SWITCH_HANDLE_TYPE_RIF <<
+                                       SWITCH_HANDLE_TYPE_SHIFT)));
+    } else if (SWITCH_LAG_HANDLE(rif_handle)) {
+      status = tdi_key_field_set_value(key_hdl, field_id,
+                                       (rif_handle &
+                                       ~(SWITCH_HANDLE_TYPE_LAG <<
+                                       SWITCH_HANDLE_TYPE_SHIFT)));
+    }
```

Comment:
better to have and else case to handle errors.
same for other places as well

Reply:
Addressed.